### PR TITLE
Bug 48234 - Unable to determine which `Application Output` pad belongs to which project

### PR DESCRIPTION
**Pull request by DavidKarlas**
This is different aproach from https://github.com/mono/monodevelop/pull/1697
It sets suffix when Application Output is created and not from debugger when process is started/connected to.

Id of pad needs to be suffixed with title otherwise on reuse title can be wrong. Only downside of this is that we can end with many open Application Output pads because we don't reuse pads with different titles.

### CODE REVIEW

**Comment by slluis** [comment]
We have to reuse pads with different titles. For example, if I a solution has projects A and B, if I start A, stop, then start B, it should reuse the same output pad.

**Comment by DavidKarlas** [comment]
@slluis I pushed new commit that removes need for different Ids by doing checking if pads have same title and finding smallest free " (#)" suffix

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/Pad.cs [hunk 1]
```diff
@@ -63,6 +63,7 @@ internal Pad (DefaultWorkbench workbench, PadCodon content)
 		
 		public string Title {
 			get { return window.Title; }
+			set { window.Title = value; }
```
Comment (slluis) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/Pad.cs]: Thist setter should not be public. The implementation of a pad has api for changing the title.

**Comment by DavidKarlas** [comment]
@slluis I updated commit, can you check if it's OK now?

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/ProgressMonitors.cs [hunk 2]
```diff
@@ -78,9 +78,9 @@ private ProgressMonitor GetBuildProgressMonitor (string statusText)
 			return mon;
 		}
 		
-		public OutputProgressMonitor GetRunProgressMonitor ()
+		public OutputProgressMonitor GetRunProgressMonitor (string titleSuffix = null)
```
Comment (slluis) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/ProgressMonitors.cs]: This is an abi break. Please add a new overload.

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/ProgressMonitors.cs [hunk 3]
```diff
@@ -142,8 +142,11 @@ public OutputProgressMonitor GetOutputProgressMonitor (string title, IconId icon
 			return GetOutputProgressMonitor (null, title, icon, bringToFront, allowMonitorReuse, visible);
 		}
 		
-		public OutputProgressMonitor GetOutputProgressMonitor (string id, string title, IconId icon, bool bringToFront, bool allowMonitorReuse, bool visible = true)
+		public OutputProgressMonitor GetOutputProgressMonitor (string id, string title, IconId icon, bool bringToFront, bool allowMonitorReuse, bool visible = true, string titleSuffix = null)
```
Comment (slluis) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/ProgressMonitors.cs]: another abi break

**Comment by DavidKarlas** [comment]
@slluis I fixed ABI breaking change, see any other issue?

**Comment by slluis** [comment]
The behavior is not yet correct. Let's say you have a solution with projects A and B. Issues I found:

* Run project A. The title of the output channel is changed to A. Wait for it to end. Now run A again. The title of the pad changes to A (1). It should not change.
* Run project A. While A is running, start project B. Pads with titles A and B are created. That's correct. Now start B again. Instead of reusing the pad with title B that already exists, it uses the pad that had title A and changes it to B (2).

**Comment by DavidKarlas** [comment]
@slluis any more comment on this PR? I fixed both issues.

**Comment by slluis** [comment]
It looks good now. Please merge to master too.

**Pull request merged**